cupy/cuda/cudnn.pyx

```diff
@@ -640,6 +657,48 @@ cpdef addTensor_v3(size_t handle, size_t alpha, size_t bDesc,
     check_status(status)
 
 
+###############################################################################
+# Tensor operations
+###############################################################################
+
+cpdef size_t createOpTensorDescriptor() except *:
+    cdef OpTensorDescriptor opTensorDesc
+    status = cudnnCreateOpTensorDescriptor(&opTensorDesc)
+    check_status(status)
+    return <size_t>opTensorDesc
+
+cpdef setOpTensorDescriptor(size_t opTensorDesc, int opTensorOp,
+                            int opTensorCompType, int opTensorNanOpt):
+    status = cudnnSetOpTensorDescriptor(
+        <OpTensorDescriptor>opTensorDesc, <OpTensorOp>opTensorOp,
+        <DataType>opTensorCompType, <NanPropagation>opTensorNanOpt)
+    check_status(status)
+
+cpdef getOpTensorDescriptor(size_t opTensorDesc):
+    cdef OpTensorOp opTensorOp
+    cdef DataType opTensorCompType
+    cdef NanPropagation opTensorNanOpt
+    status = cudnnGetOpTensorDescriptor(
+        <OpTensorDescriptor>opTensorDesc, &opTensorOp, &opTensorCompType,
+        &opTensorNanOpt)
+    check_status(status)
+    return opTensorOp, opTensorCompType, opTensorNanOpt
+
+cpdef destroyOpTensorDescriptor(size_t opTensorDesc):
+    status = cudnnDestroyOpTensorDescriptor(<OpTensorDescriptor>opTensorDesc)
+    check_status(status)
+
+cpdef opTensor(size_t handle, size_t opTensorDesc, size_t alpha1,
+               size_t aDesc, size_t A, size_t alpha2, size_t bDesc,
+               size_t B, size_t beta, size_t cDesc, size_t C):
+    status = cudnnOpTensor(
```

Comment:
Do we need `setStream` (and thus `with nogil`) here?

Reply:
yes we do. Adding that now.